test/_detail_number.cpp

```diff
@@ -16,6 +16,8 @@
 
 #include "test_suite.hpp"
 
+#define BOOST_JSON_DETAIL_NUMBER_DEMAND_PERFECT_DOUBLE_CONVERSION 0
```

Comment:
Not sterile, and this macro is only visible in tests so you can just use `#define ACCURATE_CONVERSION` or something like that